aten/src/ATen/native/README.md

```diff
@@ -264,64 +264,77 @@ dispatch:
 
 This specifies the actual name of the function you want to dispatch to, so you
 can dispatch to different functions depending on which backend the passed tensors
-belong to.  Technically, it is also possible to write `dispatch: func_name`
-to unconditionally dispatch to a native function whose name is different than
-the name in the public ATen API, but this is generally frowned upon (just name
-them the same thing!)
+belong to.  If the dispatch table is omitted, we assume a default dispatch
+table:
+
+```
+func: func(...) -> ...
+dispatch:
+    CompositeImplicitAutograd: func
+
+# out functions get suffixed with _out in their name
+func: func.out(...) -> ...
+dispatch:
+    CompositeImplicitAutograd: func_out
```

Comment:
Hm, we don't support autograd for `out` functions, so this is pretty confusing

Reply:
Hmm, I agree, and I'm not sure how to fix this problem.

Reply:
Actually, I think this is OK. Let's take `dstack.out` as an example out function that has an implicit autograd composite registered for it. Its implementation looks like this:

```
Tensor& dstack_out(Tensor& result, TensorList tensors) {
  TORCH_CHECK(tensors.size() > 0,
           "dstack expects a non-empty TensorList");
  auto rep = at::atleast_3d(tensors);
  return at::cat_out(result, rep, 2);
}
```

This is fine; if `cat_out` had supported autograd, this would too. Similar to how composite functions support all backends their nested functions support, composite out functions support autograd if all their nested functions support autograd. OFC, right now they don't, but maybe this eventually won't be true.

Reply:
Yeah, it's not a problem functionally, it's just confusing in the docs.